src/Components/Updater/Provider.php

```diff
@@ -67,19 +69,27 @@ public function register(): void
             $this->app->singleton(Updater::class, function () use ($build) {
                 $updater = new PharUpdater($build->getPath(), false, PharUpdater::STRATEGY_GITHUB);
 
-                $composer = json_decode(file_get_contents(base_path('composer.json')), true);
-                $name = $composer['name'];
+                $composer = json_decode(file_get_contents($this->app->basePath('composer.json')), true);
```

Comment:
I think there should be a way to set the package name without Composer, as there is no real guarantee that the compiled binary is run in a directory with a `composer.json` file, nor do I think there should need to be?

Reply:
This is the `composer.json` within the Phar file. It should always exist.

Reply:
Okay, that's weird, because for me, it tries to load it from the working directory. 
https://github.com/hydephp/cli/commit/925f00b011211bb6590169f5a1e0cb5308d04057 It could be due to how my paths are setup though, in which case this is my fault. Gonna check it out.

Reply:
Yeah, this is my bad. My application sets the working base path to be the working directory. For my usages, I would need this to be an absolute path, but that is probably not something Zero should worry about.

Reply:
Update: Changing the line to the following works better for me.

```php
$composer = json_decode(file_get_contents(__DIR__.'/../../../../../../composer.json'), true);
```